### PR TITLE
[FEAT] Add str.left() funtion

### LINKED ISSUE #1926: [EXPRESSIONS] `.str.left`
`.str.left(n)`

Return the n left-most characters.

https://ibis-project.org/reference/expression-strings#ibis.expr.types.strings.StringValue.left

### PR DESCRIPTION
**Pull request by murex971**
Returns the left-most n characters

Resolves #1926

### CODE REVIEW

File: tests/series/test_utf8_ops.py
Comment (colin-ho) [inline tests/series/test_utf8_ops.py:1]: nit: rename `pattern` to `nchars` the argument to `str.left` is not really a pattern but more of a number representing the desired number of chars

File: src/daft-dsl/src/functions/utf8/left.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    fn fn_name(&self) -> &'static str {
+        "left"
+    }
+
+    fn to_field(&self, inputs: &[Expr], schema: &Schema, _: &Expr) -> DaftResult<Field> {
+        match inputs {
+            [data, pattern] => match (data.to_field(schema), pattern.to_field(schema)) {
+                (Ok(data_field), Ok(pattern_field)) => {
+                    match (&data_field.dtype, &pattern_field.dtype) {
+                        (DataType::Utf8, DataType::UInt32) => {
```
Comment (colin-ho) [inline src/daft-dsl/src/functions/utf8/left.rs:24]: We should also support other integer datatypes, such as UInt64, Int32, Int64, etc.

File: src/daft-core/src/series/ops/utf8.rs [hunk 2]
```diff
@@ -119,4 +119,14 @@ impl Series {
             ))),
         }
     }
+
+    pub fn utf8_left(&self, pattern: &Series) -> DaftResult<Series> {
+        match self.data_type() {
+            DataType::Utf8 => Ok(self.utf8()?.left(pattern.u32()?)?.into_series()),
```
Comment (colin-ho) [inline src/daft-core/src/series/ops/utf8.rs:125]: Likewise, support other integer datatypes.

File: src/daft-core/src/array/ops/utf8.rs [hunk 3]
```diff
@@ -260,6 +260,81 @@ impl Utf8Array {
         Ok(Utf8Array::from((self.name(), Box::new(arrow_result))))
     }
 
+    pub fn left(&self, n: &UInt32Array) -> DaftResult<Utf8Array> {
```
Comment (colin-ho) [inline src/daft-core/src/array/ops/utf8.rs:263]: You can take advantage of our `DaftIntegerType` trait https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/datatypes/mod.rs#L305, to make this a generic function. example usage here: https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/array/ops/sort.rs#L67-L70

**Review by colin-ho — COMMENTED** [review]
Hey @murex971, aweome work! In addition to the comments I left, could you also add a test in `tests/table/utf8` similar to this one https://github.com/Eventual-Inc/Daft/blob/main/tests/table/utf8/test_match.py. Thanks :)

**Comment by murex971** [comment]
@colin-ho  pushed the changes.

File: daft/expressions/expressions.py [hunk 4]
```diff
@@ -854,6 +854,18 @@ def capitalize(self) -> Expression:
         """
         return Expression._from_pyexpr(self._expr.utf8_capitalize())
 
+    def left(self, pattern: int | Expression) -> Expression:
```
Comment (colin-ho) [inline daft/expressions/expressions.py:857]: Just one last nit here, could you rename this (and all the other occurrences of `pattern`) to `nchars` like you did for the tests? Just for consistency 😄 , after this it should be good to merge!

**Comment by murex971** [comment]
@colin-ho thanks for the review. Addressed.

**Review by colin-ho — APPROVED** [review]

**Comment by colin-ho** [comment]
> @colin-ho thanks for the review. Addressed.

Awesome, thanks!

**Pull request merged**